src/averageAges.js

```diff
@@ -36,8 +45,20 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(person =>
+      person.sex === 'f' && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === FEMALE);
```

Comment:
When checking whether a mother has children for the calculateWomenAverageAge function with `withChildren` parameter, the approach efficiently filters, but be cautious with scalability. High-complexity nested iterations can largely impact performance in datasets with many people.